worldedit-core/src/main/java/com/boydti/fawe/object/collection/FastRandomCollection.java

```diff
@@ -5,28 +5,41 @@
 
 import java.util.ArrayList;
 import java.util.Map;
+import java.util.Optional;
 
 public class FastRandomCollection<T> extends RandomCollection<T> {
-    private T[] values;
+    private final T[] values;
 
-    public FastRandomCollection(Map<T, Double> weights, SimpleRandom random) {
-        super(weights, random);
+    private FastRandomCollection(T[] values, SimpleRandom random) {
+        super(random);
+        this.values = values;
+    }
+
+    /**
+     * Create a new FastRandomCollection if the given values and weights match the criteria.
+     * @param weights the weight of the values.
+     * @param random the random generator to use for this collection.
+     * @param <T> the value type.
+     * @return an {@link Optional} containing the new collection if it could
+     * be created, {@link Optional#empty()} otherwise.
+     */
+    public static <T> Optional<RandomCollection<T>> create(Map<T, Double> weights, SimpleRandom random) {
         int max = 0;
         int[] counts = new int[weights.size()];
         Double[] weightDoubles = weights.values().toArray(new Double[0]);
         for (int i = 0; i < weightDoubles.length; i++) {
             int weight = (int) (weightDoubles[i] * 100);
             counts[i] = weight;
             if (weight != (weightDoubles[i] * 100)) {
-                throw new IllegalArgumentException("Too small");
+                return Optional.empty();
             }
             if (weight > max) {
                 max = weight;
             }
         }
         int gcd = MathMan.gcd(counts);
         if (max / gcd > 100000) {
-            throw new IllegalArgumentException("Too large");
+            return Optional.empty();
```

Comment:
Instead of just returning empty, maybe a message here would be nice?

Reply:
I tried to improve the previous implementation. From what I've seen, the exceptions where thrown to exit the construtor because the object couldn't be created and the exception was basically ignored. `FastRandomCollection` should be used internally by `RandomCollection` only, means the API is `RandomCollection.of(...)` and it returns the implementation based on the given values. Why the `FastRandomCollection` couldn't be used isn't relevant to the user.

That said, it might be worth to point out proper usage of the RandomCollection API. The reason I didn't documented the cases in which a FastRandomCollection can be created is that I didn't deeply looked into the algorithm (I'm especially confused by the `100000` value but who knows why it is like that).

Reply:
Ok, in that case it's probably fine.